aioxmpp/muc/service.py

```diff
@@ -618,18 +633,28 @@ def _match_tracker(self, message):
         try:
             tracker = self._tracking_by_id[message.id_]
         except KeyError:
+            key = message.from_, _extract_one_pair(message.body)
```

Comment:
Note, the runtime of the version using the order is also in some way O(n·m) (n ... number of the messages, m ... number of language entries per message), but all discussed schemes are only O(N) where (N ... size of received messages).